pybamm/solvers/scikits_dae_solver.py

```diff
@@ -77,3 +97,38 @@ def rootfn(t, y, ydot, return_root):
 
         # return solution, we need to tranpose y to match scipy's interface
         return sol.values.t, np.transpose(sol.values.y)
+
+    def auto_jac(self, residuals):
+        """
+        Compute Jacobian of DAE model using autograd.
+
+        Parameters
+        ----------
+        residuals : method
+            A function that takes in t, y and ydot and returns the residuals of the
+            equations
+
+        """
+        self.jacobian_ydot = autograd.jacobian(residuals, 2)
+        self.jacobian_rhs_alg = autograd.jacobian(residuals, 1)
+        return self.jacobian_ydot, self.jacobian_rhs_alg
```

Comment:
python functions should either assign a class attribute or return a result, doing both is potentially confusing

Reply:
Thanks for your help, changing to `set_auto_jac` and removing `jacobian` as discussed. I'll also add jacobian calculation to the other solvers